src/test/java/org/cryptomator/windows/autostart/WinShortcutCreationTest.java

```diff
@@ -0,0 +1,42 @@
+package org.cryptomator.windows.autostart;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class WinShortcutCreationTest {
+
+	private Path linkTarget;
+	private Path shortcut;
+
+	@BeforeEach
+	public void setup(@TempDir Path tempDir) throws IOException {
+		this.linkTarget = tempDir.resolve("link.target");
+		Files.createFile(linkTarget);
+	}
+
+	@Test
+	public void testShellLinkCreation(@TempDir Path tempDir) {
+		WinShortcutCreation winShortcutCreation = new WinShortcutCreation();
+		shortcut = linkTarget.getParent().resolve("short.lnk");
+
+		int returnCode = winShortcutCreation.createShortcut(linkTarget.toString(), shortcut.toString(), "asd");
+
+		Assertions.assertEquals(0, returnCode);
+		Assertions.assertTrue(Files.exists(shortcut));
```

Comment:
maybe assert that this is _not_ a regular file etc?

Reply:
Windows shortcuts are regular files. They have a special meaning in the windows shell/explorer, but when it comes down to the filesystem level, a shortcut is just a plain old file object.